File: source_funcs/go_transcoder/SEARCHING_ARRAY_ADJACENT_DIFFER_K/SEARCHING_ARRAY_ADJACENT_DIFFER_K_emit.go
package main

import (
	"fmt"
	"math"
)

func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}

//go:export f_gold
func f_gold(arr []int, n int, x int, k int) int {
	var i int = 0
	for i < n {
		if arr[i] == x {
			return i
		}
		i = i + max(1, int(math.Abs(float64(arr[i]-x))/float64(k)))
	}
	fmt.Print("number is not present!")
	return -1
}

File: rust_funcs/go_transcoder/SEARCHING_ARRAY_ADJACENT_DIFFER_K/SEARCHING_ARRAY_ADJACENT_DIFFER_K_emit.rs
#![allow(dead_code)]
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(non_snake_case)]
#![allow(non_camel_case_types)]
#![allow(improper_ctypes_definitions)]
#[no_mangle]
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
#[no_mangle]
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
#[no_mangle]
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
#[no_mangle]
fn len(arr: &[i32]) -> usize {arr.len()}
#[no_mangle]
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32;

#[no_mangle]
fn f_gold(arr: [i32;2], n: i32, x: i32, k: i32) -> i32 {
  let mut i: f32 = 0.;
  while i < n as f32{
    if arr[i as usize] == x {return i as i32;}
    i = i + f32::max(1., (arr[i as usize] - x) as f32 / k as f32);
  }
  println!("number is not present!");
  return -1; 
}
